src/webloader.js

```diff
@@ -123,6 +123,35 @@ function checkLazyUid(script) {
 	return false;
 }
 
+function requireConsent(script) {
+  return script.dataset.evolvRequireConsent === 'true'
+}
+
+let consented = false;
+let localConsentStore = {};
+let localSessionConsentStore = {};
+
+function markConsented(evolv) {
+  if (consented) return;
+  consented = true;
+
+  Object.entries(localConsentStore).forEach(function (entry) {
+    window.evolv.store(entry[0], entry[1].value, false);
+  });
+
+  Object.entries(localSessionConsentStore).forEach(function (entry) {
+    window.evolv.store(entry[0], entry[1].value, true);
+  });
+
+  localConsentStore = {};
+  localSessionConsentStore = {};
+  evolv.client.allowEvents();
+}
+
+function hasConsented() {
+  return consented;
+}
+
```

Comment:
For what it's worth, I do think this file is becoming somewhat of a god class. It may worth breaking out some of this behavior into a separate file/object.